src/main/java/com/example/demo/service/ProductService.java

```diff
@@ -22,93 +24,117 @@
-    }
-
-    public UUID attach(AttachmentCreateDto attachmentCreateDto) {
-        var attachmentDomain = attachmentMapper.fromDto(attachmentCreateDto);
-        var product = productRepository.findById(attachmentDomain.getProductId());
-        return product.map(p -> {
-            if (!p.getAttachments().stream().map(AttachmentDomain::getAttachment).toList().contains(attachmentDomain.getAttachment())) {
-                return attachmentRepository.save(attachmentDomain).getId();
-            } else
-                return null;
-        }).orElse(null);
-    }
-
-    public Integer removeAttachment(UUID id) {
-        var att = attachmentRepository.findById(id);
-        return att.map(a -> {
-            attachmentRepository.deleteById(id);
-            return 1;
-        }).orElse(0);
-    }
+	private final ProductMapper productMapper;
+	private final CriteriaMapper criteriaMapper;
+	private final AttachmentMapper attachmentMapper;
+
+	private final SupplierRepository supplierRepository;
+
+	private final CategoryRepository categoryRepository;
+
+	private final ProductRepository productRepository;
+
+	private final AttachmentRepository attachmentRepository;
+
+	public ProductService(ProductMapper productMapper, CriteriaMapper criteriaMapper, AttachmentMapper attachmentMapper, SupplierRepository supplierRepository, CategoryRepository categoryRepository, ProductRepository productRepository, AttachmentRepository attachmentRepository) {
+		this.productMapper = productMapper;
+		this.criteriaMapper = criteriaMapper;
+		this.attachmentMapper = attachmentMapper;
+		this.supplierRepository = supplierRepository;
+		this.categoryRepository = categoryRepository;
+		this.productRepository = productRepository;
+		this.attachmentRepository = attachmentRepository;
+	}
+
+	public List<ProductReadDto> viewProducts() {
+		var products = productRepository.findAll();
+		return StreamSupport.stream(products.spliterator(), false).map(productMapper::toDto).collect(Collectors.toList());
+	}
+
+
+	private void checkProductUniqueness(ProductDomain productDomain) {
+		var supplierDomain = supplierRepository.findById(productDomain.getSupplier().getId());
+		if (supplierDomain.isEmpty()) {
+			throw new NotFoundException("Supplier with id " + productDomain.getSupplier().getId() + " not found.");
+		}
+		var categoryDomain = categoryRepository.findById(productDomain.getCategory().getId());
+		if (categoryDomain.isEmpty()) {
+			throw new NotFoundException("Category with id " + productDomain.getCategory().getId() + " not found.");
+		}
+	}
+
+	public ProductReadDto addProduct(ProductCreateDto productCreateDto) {
+		var productDomain = productMapper.fromCreateDto(productCreateDto);
+		if (productDomain == null){
+			throw new MappingException("Product mapping error");
+		}
```

Comment:
Very similar to 

var productDomain = productMapper.fromUpdateDto(productUpdateDto);
		if (productDomain == null){
			throw new MappingException("Product mapping error");
		}